packages/eslint-plugin/configs/jsdoc.js

```diff
@@ -93,6 +93,7 @@ module.exports = {
 					...temporaryWordPressInternalTypes,
 					...temporaryExternalTypes,
 					'void',
+					'JSX',
```

Comment:
JSX is in global namespace. That's why it is added here.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7d19d6ff87c33d30e2c81b44e5763f7d0453233f/types/react/index.d.ts#L2943-L2944

Reply:
Do you know at what point this would come "in scope" ? I guess if it were an issue, the TypeScript build would be complaining about it?

Reply:
The members of JSX namespace are all types or interfaces. They're imported when we import `react`. So, TypeScript will only complain we use JSX elements before `import something from 'react'` like:

```js
/**
* @return {JSX.Element} a component
*/
function f() {}

const React = require('react');
```

I'm not sure it would be even an error if hoisting happens. It's also against our coding convention. 

In conclusion, it's almost impossible to happen.

Reply:
One of the worries I might have is that for `createElement` in particular, we automate this via a Babel transform, so that the source doesn't actually include an `import` statement.

See: https://github.com/WordPress/gutenberg/tree/master/packages/babel-plugin-import-jsx-pragma

Reply:
I've tested by adding a new file without importing `React`:

```js
// a11y/src/index.js

export { X } from './x';
```

```js
// a11y/src/x.js

/**
 * Test if it works without TypeScript complaining.
 *
 * @return {JSX.Element} React component
 */
export const X = () => <div />;
```

It works without any complaints. We're good to go.